71.md

```diff
@@ -0,0 +1,121 @@
+* `title` (required) title of the video
+* `"published_at"`, for the timestamp in unix seconds (stringified) of the first time the video was published
+* `summary` (optional) summary/description of the video (same as `.content`)
+* `"aes-256-gcm"` (optional)  key and nonce for AES-GCM encryption with tagSize always 128bits
+* `x` containing the SHA-256 hexencoded string of the file.
+* `size` (optional) size of file in bytes
+* `dim` (optional) size of file in pixels in the form `<width>x<height>`
+* `duration` (optional) video duration in seconds
+* `magnet` (optional) URI to magnet file
+* `i` (optional) torrent infohash
+* `text-track` (optional, repeated) link to WebVTT file for video, type of supplementary information (captions/subtitles/chapters/metadata), optional language code
+* `thumb` (optional) url of thumbnail with same aspect ratio
+* `image` (optional) url of preview image with same dimensions
+* `content-warning` (optional) warning about content of NSFW video
+* `alt` (optional) description for accessibility
+* `segment` (optional, repeated) start timestamp in format `HH:MM:SS.sss`, end timestamp in format `HH:MM:SS.sss`, chapter/segment title, chapter thumbnail-url
+* `t` (optional, repeated) hashtag to categorize video
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant in the video, optional recommended relay URL
+* `r` (optional, repeated) references / links to web pages
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": 34235 | 34236,
+  "content": "<summary / description of video>",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["title", "<title of video>"],
+    ["summary", "<summary of video>"],
+    ["thumb", "<thumbnail image for video>"],
+    ["published_at", "<unix timestamp>"],
+    ["alt", <description>],
+
+    // Video Data
+    ["url",<string with URI of file>],
+    ["m", <MIME type>],
+    ["x",<Hash SHA-256>],
+    ["aes-256-gcm",<key>, <iv>],
+    ["size", <size of file in bytes>],
+    ["duration", <duration of video in seconds>],
+    ["dim", <size of file in pixels>],
+    ["magnet",<magnet URI> ],
+    ["i",<torrent infohash>],
+    ["text-track", "<encoded `kind 6000` event>", "<recommended relay urls>"],
+    ["content-warning", "<reason>"],
+    ["segment", <start>, <end>, "<title>", "<thumbnail URL>"],
+
+    // Participants
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+
+    // Hashtags
+    ["t", "<tag>"],
+    ["t", "<tag>"],
+
+    // Reference links
+    ["r", "<url>"],
+    ["r", "<url>"]
+  ]
+}
+```
+
+## Video View
```

Comment:
This is basically the same intent as @staab 's https://github.com/nostr-protocol/nips/pull/933

I think we should just use `kind:15` here too with a `[ "range", "<start>", "<end>" ]`

Reply:
I think a NIP-33 is a bit of an overkill here.

Reply:
> This is basically the same intent as @staab 's https://github.com/nostr-protocol/nips/pull/933
> 
> 
> 
> I think we should just use `kind:15` here too with a `[ "range", "<start>", "<end>" ]`

I think that it would be useful to have both a kind 15 "Seen event" and an independent "View event". The main purpose of the kind 15 would be to clear notifications which makes sense in the context of checking your feed of subscriptions to see if anything new has been posted. But the kind 34237 is intended to indicate a video view, to both allow for users to continue from where they left off as well as allow clients to show a YouTube-like view count for a particular video. This may be useful for creating algorithmic feeds down the line.